choice.rb

```diff
@@ -0,0 +1,96 @@
+require './student'
+require './teacher'
+require './book'
+require './rental'
+
+class HandleMenuChoice
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_books
+    if @books.empty?
+      puts 'There is no book registered in the library'
+    else
+      @books.each { |book| puts book }
+    end
+  end
+
+  def list_people
+    if @people.empty?
+      puts 'There is no one registered in the library'
+    else
+      @people.each { |person| puts person }
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    option = gets.chomp
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    case option
+    when '1'
+      create_student(age, name)
+    when '2'
+      create_teacher(age, name)
+    else
+      puts 'Not a valid option'
+      return
+    end
+    puts 'Person created successfully'
+  end
+
+  def create_student(age, name)
+    print 'Has parent permission? [Y/N]: '
+    pp = gets.chomp
+    @people.push(Student.new(name: name, age: age, parent_permission: translate_answer(pp.downcase)))
+  end
+
+  def create_teacher(age, name)
+    print 'Specialization: '
+    specialization = gets.chomp
+    @people.push(Teacher.new(name: name, age: age, specialization: specialization))
```

Comment:
- [ ] This line (Teacher.new(name: name, age: age, parent_permission: translate_answer(pp.downcase))) should've been placed in the 'create_person()'`
> A client should never be forced to implement an interface that it doesn’t use, or clients shouldn’t be forced to depend on methods they do not use.